### DISCUSSION TITLE
DTO의 패키지 분류

### DISCUSSION THREAD

**Discussion by yxxnghwan** [woowacourse-teams/2022-nae-pyeon#22] [BE 컨벤션 이슈]
현재 구조에서 패키지가 `controller.dto`와 `service.dto` 두 패키지서 DTO를 관리고 있는데요!
DTO의 사용 범위를 각각의 레이어마다 나눌 건지, 아니면 데이터 모양이 같은 DTO의 경우 여러 레이어에서 같이 사용할 수 있도록 설계할 건지에 대해서 토론해보면 좋을 것 같아요!

해당 결과에 대해서 dto를 관리하는 패키지 구조가 변경될 수 있을 것 같아요.
각자 자유롭게 의견 부탁드립니다!🤗

**Comment by kth990303** [comment]
저는 dto package만 있으면 된다고 생각하긴 했습니다.
service dto, controller dto를 따로 패키지를 분리해서 다루는 건 오히려 불필요한 분리라고 생각했어요.

컨트롤러에서 request dto를 보내주면, service에서 response dto를 보내주고 그 dto를 컨트롤러에서 응답으로 넘겨주는 단순한 개념으로 생각하면 dto package만 있으면 된다고 생각했습니다!

**Comment by Seungpang** [comment]
저는 둘다 상관없는 입장이긴합니다.
다만 controller에서 받은 request가 service까지 내려가는걸 싫어해서 다 까서 넘겨주는 방향이지만 (여러 컨트롤러에서 해당 서비스를 사용할 수 있게) 서비스 레이어에 맞게 새로운 dto를 만들어서 사용해보는것도 괜찮아 보여서 하게 되었습니다. 다만 무의미한 중복이 발생할 수 있다는게 문제로 보이긴 하네요

**Comment by asebn1** [comment]
그동안 진행한 프로젝트에서는 controller의 dto와 service의 dto를 구분하긴 했지만 필드가 달랐던 적(사용되는 시점이 다른 dto)이 없어서 코드 중복이 발생한다는 점에서 불필요할 수 있다는 것에 동의합니다.
다만 저도 승팡처럼 dto를 레이어로 분리하지 않았을 때, controller, service간의 의존방향이 달라질 수 있다는 문제가 크다고 느껴 설계관점에서는 나누는게 좋겠다고 생각했어요.
둘 다 장단점이 있다고 생각해서 두 가지 방법 다 좋습니다.

https://umbum.dev/1206
http://tech.pick-git.com/%EB%B3%80%EA%B2%BD%EC%97%90-%EC%9C%A0%EC%97%B0%ED%95%9C-%EC%84%A4%EA%B3%84,-%EC%9A%B0%EB%A6%AC%EB%8A%94-2%EA%B0%80%EC%A7%80-%EC%A2%85%EB%A5%98%EC%9D%98-DTO%EB%A5%BC-%EC%93%B4%EB%8B%A4/
